app/src/main/java/com/github/se/studybuddies/viewModels/CalendarViewModel.kt

```diff
@@ -1,23 +1,106 @@
 package com.github.se.studybuddies.viewModels
 
+import androidx.compose.runtime.mutableStateMapOf
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.github.se.studybuddies.data.CalendarDataSource
 import com.github.se.studybuddies.data.CalendarUiState
+import com.github.se.studybuddies.data.DailyPlanner
+import com.github.se.studybuddies.database.DatabaseConnection
 import java.time.YearMonth
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
+class CalendarViewModel(private val uid: String) : ViewModel() {
+
+  private val dataSource by lazy { CalendarDataSource() }
+  private val databaseConnection = DatabaseConnection()
+
+  private val _dailyPlanners = mutableStateMapOf<String, MutableStateFlow<DailyPlanner>>()
+  val dailyPlanners: Map<String, StateFlow<DailyPlanner>>
+    get() = _dailyPlanners
+
+  private val _uiState = MutableStateFlow(CalendarUiState.Init)
+  val uiState: StateFlow<CalendarUiState> = _uiState.asStateFlow()
+
+  init {
```

Comment:
From what the coach told us on the general chat on discord, using init in a viewModel is a bad practice and should be avoided. I'll let you have a look at our conversation on discord.

Reply:
will change it in another pr